File: abc/abc121/abc121_b.cs
// https://atcoder.jp/contests/abc121/submissions/9279460

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            var N = input[0];
            var M = input[1];
            var C = input[2];

            var B = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

            var A = new int[N][];
            for (int i = 0; i < N; i++)
            {
                A[i] = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            }

            var result = 0;
            for (int i = 0; i < N; i++)
            {
                var temp = A[i].Zip(B, (a, b) => a * b).Sum();
                result = temp + C > 0 ? ++result : result;
            }

            Console.WriteLine(result);
        }
    }
}

File: abc/abc121/abc121_b.fs
// https://atcoder.jp/contests/abc121/submissions/12277754

let [| N; M; C |] = stdin.ReadLine().Split() |> Array.map int
let B = stdin.ReadLine().Split() |> Array.map int

let rec solver r i =
    if i = N then r
    else
        let A = stdin.ReadLine().Split() |> Array.map int

        let s =
            Array.zip A B
            |> Array.map (fun x -> (fst x) * (snd x))
            |> Array.sum
        if s + C > 0 then solver (r + 1) (i + 1)
        else solver r (i + 1)

solver 0 0 |> stdout.WriteLine
